### PR TITLE
BUG - Add missing userId field to data models

### PR DESCRIPTION
**Pull request by 1jeanpaul1**
## Problem
userId is needed in the Post and Comment database tables but it was not included

## Solution
Add userId

## Test
1. Pull down this branch
2. **Run and Build:** `docker-compose up --build`
3. Go to the following endpoint and confirm that it returns **userId** data now:
http://localhost:8082/forum/post/v1/all (Will contain post data populated in `DataPopulation.java`)
4. Go to http://localhost:9000/ and confirm the `Post` and `Comment` tables have a **userId** field

--
Ticket: https://trello.com/c/gNYJfKAV/22-bug-posts-and-comments-dont-have-a-user-reference

### CODE REVIEW

File: src/main/java/com/fcgl/madrid/forum/model/Post.java [hunk 1]
```diff
@@ -30,7 +31,8 @@
     public Post() {
```
Comment (Mikerad1) [inline src/main/java/com/fcgl/madrid/forum/model/Post.java:31]: Think this should be removed as we should force the setting of the values when creating a Post. What do you think?

File: src/main/java/com/fcgl/madrid/forum/model/Comment.java [hunk 2]
```diff
@@ -90,4 +92,12 @@ public Post getPost() {
     public void setPost(Post post) {
         this.post = post;
     }
+
+    public int getUserId() {
+        return userId;
+    }
+
+    public void setUserId(int userId) {
```
Comment (Mikerad1) [inline src/main/java/com/fcgl/madrid/forum/model/Comment.java]: Think this should be removed as who made the comment should be able to change.

File: src/main/java/com/fcgl/madrid/forum/model/Comment.java [hunk 3]
```diff
@@ -17,12 +17,14 @@
     private long updatedDate;
```
Comment (Mikerad1) [inline src/main/java/com/fcgl/madrid/forum/model/Comment.java:17]: Shouldn't there be a get and set for createdDate, updatedDate, likes and dislikes?

File: src/main/java/com/fcgl/madrid/forum/model/Post.java [hunk 4]
```diff
@@ -112,4 +114,12 @@ public void setCityId(int cityId) {
     public void setComment(List<Comment> comment) {
         this.comment = comment;
     }
+
+    public int getUserId() {
+        return userId;
+    }
+
+    public void setUserId(int userId) {
```
Comment (Mikerad1) [inline src/main/java/com/fcgl/madrid/forum/model/Post.java]: Think this should be removed as the user who created the comment shouldn't change

File: src/main/java/com/fcgl/madrid/forum/model/Post.java [hunk 5]
```diff
@@ -22,6 +22,7 @@
     private int likes;
```
Comment (Mikerad1) [inline src/main/java/com/fcgl/madrid/forum/model/Post.java:22]: Shouldn't there be get and set methods for updatedDate, createdDate, likes, dislikes and cityID?

**Review by Mikerad1 — CHANGES_REQUESTED** [review]

File: src/main/java/com/fcgl/madrid/forum/model/Post.java (hunk 1 above)
Comment (fcgl) [reply src/main/java/com/fcgl/madrid/forum/model/Post.java:31]: I think this is required for Spring, can try without it and see if it works

File: src/main/java/com/fcgl/madrid/forum/model/Comment.java (hunk 2 above)
Comment (fcgl) [reply src/main/java/com/fcgl/madrid/forum/model/Comment.java]: Maybe I'll make it private? I think this is required by Spring. Could be wrong. Will try without it

File: src/main/java/com/fcgl/madrid/forum/model/Comment.java (hunk 3 above)
Comment (fcgl) [reply src/main/java/com/fcgl/madrid/forum/model/Comment.java:17]: Yes! I'll add them in

File: src/main/java/com/fcgl/madrid/forum/model/Post.java (hunk 4 above)
Comment (fcgl) [reply src/main/java/com/fcgl/madrid/forum/model/Post.java]: Same as above

File: src/main/java/com/fcgl/madrid/forum/model/Post.java (hunk 5 above)
Comment (fcgl) [reply src/main/java/com/fcgl/madrid/forum/model/Post.java:22]: Yes!!!! :)

Comment (Mikerad1) [reply src/main/java/com/fcgl/madrid/forum/model/Post.java:22]: oh sorry, didn't see it. the file viewer didn't expand everything

File: src/main/java/com/fcgl/madrid/forum/model/Comment.java (hunk 3 above)
Comment (Mikerad1) [reply src/main/java/com/fcgl/madrid/forum/model/Comment.java:17]: oh sorry, didn't see it. the file viewer didn't expand everything

File: src/main/java/com/fcgl/madrid/forum/model/Post.java (hunk 1 above)
Comment (fcgl) [reply src/main/java/com/fcgl/madrid/forum/model/Post.java:31]: `An empty constructor is needed to create a new instance via reflection by your persistence framework.`

https://stackoverflow.com/questions/18099127/java-entity-why-do-i-need-an-empty-constructor

Comment (Mikerad1) [reply src/main/java/com/fcgl/madrid/forum/model/Post.java:31]: Ok cool, but then shouldn't there be one for the comment class aswell?

Comment (fcgl) [reply src/main/java/com/fcgl/madrid/forum/model/Post.java:31]: The comment class does have one

File: src/main/java/com/fcgl/madrid/forum/model/Comment.java (hunk 2 above)
Comment (fcgl) [reply src/main/java/com/fcgl/madrid/forum/model/Comment.java]: Okay, removed any setters that we don't need

File: src/main/java/com/fcgl/madrid/forum/model/Post.java (hunk 4 above)
Comment (fcgl) [reply src/main/java/com/fcgl/madrid/forum/model/Post.java]: good call

**Comment by fcgl** [comment]
@Mikerad1 Should be ready for a re-review

File: src/main/java/com/fcgl/madrid/forum/model/Post.java (hunk 1 above)
Comment (Mikerad1) [reply src/main/java/com/fcgl/madrid/forum/model/Post.java:31]: oh sorry. completely missed it. will test it now

**Review by Mikerad1 — APPROVED** [review]
Looks good

**Pull request merged**